packages/typescript-estree/src/parser.ts

```diff
@@ -67,31 +64,22 @@ function resetExtra(): void {
  * @returns If found, returns the source file corresponding to the code and the containing program
  */
 function getASTFromProject(code: string, options: TSESTreeOptions) {
-  return firstDefined(
-    calculateProjectParserOptions(
-      code,
-      options.filePath || getFileName(options),
-      extra,
-    ),
+  const filePath = options.filePath || getFileName(options);
+  const astAndProgram = firstDefined(
+    calculateProjectParserOptions(code, filePath, extra),
     currentProgram => {
-      const ast = currentProgram.getSourceFile(
-        options.filePath || getFileName(options),
-      );
+      const ast = currentProgram.getSourceFile(filePath);
       return ast && { ast, program: currentProgram };
     },
   );
-}
 
-/**
- * @param code The code of the file being linted
- * @param options The config object
- * @returns If found, returns the source file corresponding to the code and the containing program
- */
-function getASTAndDefaultProject(code: string, options: TSESTreeOptions) {
```

Comment:
do we want to leave this in behind an option (default false) in case there are some cases that we didn't think of?

Reply:
Added the option back.